src/main/java/org/folio/search/service/reindex/ReindexOrchestrationService.java

```diff
@@ -72,8 +121,54 @@ public boolean process(ReindexRecordsEvent event) {
         event.getRangeId(), ex.getMessage()), ex);
       reindexStatusService.updateReindexMergeFailed(entityType);
       mergeRangeService.updateStatus(entityType, event.getRangeId(), ReindexRangeStatus.FAIL, ex.getMessage());
+      return true;
+    } finally {
+      // Clean up member tenant context
+      if (memberTenantId != null) {
+        ReindexContext.clearMemberTenantId();
+      }
+    }
+
+    if (reindexStatusService.isMergeCompleted()) {
```

Comment:
During my POC, i saw that merge is completed but kafka events are still being consumed causing the migration phase to be invoked multiple times. Did you encounter this?

Reply:
No, but I tested on a small amount of data, it was only 1 range for merge phase, we are planning to conduct testing on a bigger dataset later

Reply:
Is there any chance you just triggered reindex multiple times? We didn't have limitation that reindex couldn't be triggered while it's already running which caused issues in past, I only added it in this PR